rewrite-java/src/main/java/org/openrewrite/java/TypeValidation.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import org.openrewrite.java.search.FindMissingTypes;
+import org.openrewrite.java.tree.J;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TypeValidation {
+
+    private boolean classDeclarations = true;
+    private boolean identifiers = true;
+    private boolean methodDeclarations = true;
+    private boolean methodInvocations = true;
+
+    public static TypeValidation getDefault() {
+        return new TypeValidation();
+    }
+
+    public static TypeValidation none() {
+        return new TypeValidation()
+                .identifiers(false).methodInvocations(false)
+                .methodDeclarations(false).classDeclarations(false);
+    }
```

Comment:
This could be simplified with lombok. `@Data`, `@NoArgsConstructor`, `@AllArgsConstructor`, and `@Accessors(fluent = true)` would generate most of the code in this class for you

Reply:
nice suggestion thanks Sam